oaipmhTemplates/dha_RepoObject.xml

```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE RDF [
+    <!ENTITY xml "http://www.w3.org/XML/1998/namespace">
+    <!ENTITY xsd "http://www.w3.org/2001/XMLSchema#">
+    <!ENTITY owl "http://www.w3.org/2002/07/owl#">
+    <!ENTITY skos "http://www.w3.org/2004/02/skos/core#">
+    <!ENTITY dc "http://purl.org/dc/elements/1.1/">
+    <!ENTITY dct "http://purl.org/dc/terms/">
+    <!ENTITY foaf "http://xmlns.com/foaf/0.1/">
+    <!ENTITY premis "http://www.loc.gov/premis/rdf/v1#">
+    <!ENTITY dha "https://digital-humanities.at/schema#">
+    <!ENTITY ebucore "http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#">
+]>
+<rdf:RDF xmlns:owl="http://www.w3.org/2002/07/owl#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#" xmlns:xml="&xml;" xmlns:xsd="&xsd;"
+    xmlns:skos="&skos;" xmlns:dc="&dc;" xmlns:dct="&dct;" xmlns:foaf="&foaf;" xmlns:dha="&dha;" xmlns:premis="&premis;">
+
+    <!--
+    ////////////////////////////////////////////////////////////////////////////
+    //
+    // Mapping for RepoObject
+    //
+    ////////////////////////////////////////////////////////////////////////////
+    -->
+
+    <dha:RepoObject> <!-- Q: how do I include rdf:about with URI of the Resource? -->
+
+        <dha:hasMetadataDescription rdf:parseType="Resource">
+            <dc:title count="*" lang="true" val="/acdh:hasTitle"></dc:title>
+            <dc:identifier count="*" val="@acdh:hasIdentifier/acdh:hasTitle"></dc:identifier>
+            <dc:creator count="*" val="@acdh:hasCreator/acdh:hasTitle"></dc:creator>
+            <dc:contributor count="*" val="@acdh:hasContributor/acdh:hasTitle"></dc:contributor>
+        </dha:hasMetadataDescription>
+
+        <dha:basedOn> <!-- Q: is it possible to create as many elements for this property as the values for ^acdh:isSourceOf? -->
+            <dha:Object> <!-- Q: Actually, it's not sure that this is a dha:Object, and it may well be a dha:Collection. Is it possible to add some sort of if-condition? If the object is a Collection, then print dha:Collection; if it is a Resource, then print dha:Object -->
```

Comment:
There is no `if` option yet and I'm pretty skeptical regarding adding it because this leads us to the development of a completely new, fully-featured templating language which is rather not a good idea. If such stuff is needed, I would rather propose implementing a new OAI-PMH metadata format which would process XSLTs (limited to what https://www.php.net/manual/en/class.xsltprocessor.php can process so e.g. only XPath 1.0) and implement the template as an XSLT.

Reply:
Ok, so I will try to do that! Luckily, not many if-s for the moment: just this and another one for actors at line 89.